pkg/prometheus/watcher.go

```diff
@@ -470,3 +605,105 @@ func (w *Watcher) syncAlertManager(key string) error {
 	_, err = w.mClient.MonitoringV1().Alertmanagers(am.Namespace).Update(am)
 	return err
 }
+
+func (w *Watcher) handleAlertmanagerConfigAdd(obj interface{}) {
+	key, ok := keyFunc(obj)
+	if !ok {
+		return
+	}
+
+	log.Debugf("Alertmanager config added: %s", key)
+
+	enqueue(w.amcQueue, key)
+}
+
+func (w *Watcher) handleAlertmanagerConfigUpdate(oldObj, newObj interface{}) {
+	key, ok := keyFunc(newObj)
+	if !ok {
+		return
+	}
+
+	log.Debugf("Alertmanager config added: %s", key)
+
+	enqueue(w.amcQueue, key)
+}
+
+func (w *Watcher) handleAlertmanagerConfigDelete(obj interface{}) {
+	key, ok := keyFunc(obj)
+	if !ok {
+		return
+	}
+
+	log.Debugf("Alertmanager config deleted: %s", key)
+
+	enqueue(w.amcQueue, key)
+}
+
+func (w *Watcher) syncAlertManagerConfig(key string) error {
+	var alertManagerName string
+	var data []byte
+
+	log.Debugf("syncing alert manager config: key: %s", key)
+	obj, exists, err := w.amcInf.GetIndexer().GetByKey(key)
+	if err != nil {
+		return err
+	}
+
+	if !exists {
+		log.Error("Key not found...")
```

Comment:
print which key

Reply:
also, shouldn't this be treated as error?

Reply:
added the key in the error log, we are logging the error and returning, can't do much besides that, if we panic the pod will restart and monhelper is a long running process.